symbolic-debuginfo/src/elf.rs

```diff
@@ -77,10 +74,7 @@ pub struct ElfObject<'data> {
 impl<'data> ElfObject<'data> {
     /// Tests whether the buffer could contain an ELF object.
     pub fn test(data: &[u8]) -> bool {
-        matches!(
-            goblin::peek(&mut Cursor::new(data)),
-            Ok(goblin::Hint::Elf(_))
-        )
+        &data[0..elf::header::SELFMAG] == elf::header::ELFMAG
```

Comment:
subslicing can panic if you provide a `data` that is too short. Maybe use an explicit `get` here?